apps/android/LiveVideoApp/app/src/main/java/com/twilio/livevideo/app/manager/room/CameraCapturerCompat.kt

```diff
@@ -0,0 +1,135 @@
+package com.twilio.livevideo.app.manager.room
+
+import android.content.Context
+import android.graphics.ImageFormat
+import android.hardware.camera2.CameraCharacteristics
+import android.hardware.camera2.CameraManager
+import android.hardware.camera2.CameraMetadata
+import android.os.Build
+import com.twilio.video.Camera2Capturer
+import com.twilio.video.CameraCapturer
+import com.twilio.video.VideoCapturer
+import timber.log.Timber
+import tvi.webrtc.Camera1Enumerator
+import tvi.webrtc.Camera2Enumerator
+import tvi.webrtc.CameraEnumerator
+import tvi.webrtc.CapturerObserver
+import tvi.webrtc.SurfaceTextureHelper
+
+class CameraCapturerCompat(
+    private val frontCameraId: String?,
+    private val backCameraId: String?,
+    private val cameraCapturer: CameraCapturer? = null,
+    private val camera2Capturer: Camera2Capturer? = null
+) : VideoCapturer {
+
+    val cameraId: String
+        get() = cameraCapturer?.cameraId
+                ?: camera2Capturer?.cameraId
+                ?: throw IllegalStateException("At least one camera capturer must not be null")
+
+    override fun initialize(
+        surfaceTextureHelper: SurfaceTextureHelper,
+        context: Context,
+        capturerObserver: CapturerObserver
+    ) {
+        cameraCapturer?.initialize(surfaceTextureHelper, context, capturerObserver)
+                ?: camera2Capturer?.initialize(surfaceTextureHelper, context, capturerObserver)
+    }
+
+    override fun startCapture(width: Int, height: Int, framerate: Int) {
+        cameraCapturer?.startCapture(width, height, framerate) ?: camera2Capturer?.startCapture(width, height, framerate)
+    }
+
+    override fun stopCapture() {
+        cameraCapturer?.stopCapture() ?: camera2Capturer?.stopCapture()
+    }
+
+    override fun isScreencast() = cameraCapturer?.isScreencast ?: camera2Capturer?.isScreencast ?: false
```

Comment:
What is `isScreencast` for?

Reply:
This is just a method to override the behavior when in case, the capturer is like a screen casting service to external devices. You may have seen a feature to cast the screen to other Android/Google screens like TVs, but for now, this is not being used. However it's required to override from the interface `VideoCapturer` coming from `webrtc` libraries.